src/wallet/feebumper.cpp

```diff
@@ -150,12 +150,64 @@ bool TransactionCanBeBumped(const CWallet& wallet, const uint256& txid)
     if (wtx == nullptr) return false;
 
     std::vector<bilingual_str> errors_dummy;
-    feebumper::Result res = PreconditionChecks(wallet, *wtx, errors_dummy);
+    feebumper::Result res = PreconditionChecks(wallet, *wtx, /* require_mine=*/ true, errors_dummy);
     return res == feebumper::Result::OK;
 }
 
+struct SignatureWeights
+{
+    int m_sigs_count{0};
+    int64_t m_sigs_weight{0};
+
+    void AddSigWeight(size_t weight, SigVersion sigversion)
+    {
+        switch (sigversion) {
+        case SigVersion::BASE:
+            m_sigs_weight += weight * WITNESS_SCALE_FACTOR;
+            m_sigs_count += 1 * WITNESS_SCALE_FACTOR;
+            break;
+        case SigVersion::WITNESS_V0:
+            m_sigs_weight += weight;
+            m_sigs_count++;
+            break;
+        case SigVersion::TAPROOT:
```

Comment:
In "bumpfee: extract weights of external inputs when bumping fee" (a95d41bdf366939e4d3ff6368c384e752d745648)

Do we never need to do this for witness v1 outputs because these signatures are always 64 bytes so there is no need to estimate? If this is the case, would I be correct in saying that CalculateMaximumSignedInputSize() doesn't return -1 for taproot outputs in this case?

Reply:
Yes and yes.

Reply:
It would have really helped me if you had explained a bit more here.

Reply:
Wouldn't it be possible that you're creating a replacement transaction for which you have a different spending condition that you can solve than the transaction that you are replacing? E.g. you're using a scriptpath and the original transaction was a keypath spend?

Reply:
> Wouldn't it be possible that you're creating a replacement transaction for which you have a different spending condition that you can solve than the transaction that you are replacing? E.g. you're using a scriptpath and the original transaction was a keypath spend?

Yes, but we can't know that. Either the wallet is the one spending the input and so we are able to estimate the size using whatever the wallet will sign with, or the input is external and we only know the spending path that exists in the transaction to be bumped.